src/cgame/rocket/rocketChatField.h

```diff
@@ -284,6 +284,10 @@ class RocketChatField : public Rml::Element, Rml::EventListener
 
 	void MoveCursorForward()
 	{
+		if (cursor_character_index >= static_cast<size_t>( Q_UTF8_Strlen( text.c_str() ) ) )
```

Comment:
cursor_character_index is actually just a byte index. So you want text.size() not Q_UTF8_Strlen.

Reply:
Ahh... I thought that cursor_character_index was the index of the character. I didn't look at the code closely enough. Thanks.